x-pack/plugins/endpoint/server/routes/resolver/queries/base.ts

```diff
@@ -15,11 +21,14 @@ export abstract class ResolverQuery {
     private readonly pagination?: PaginationParams
   ) {}
 
-  protected paginateBy(field: string, query: JsonObject) {
-    if (!this.pagination) {
-      return query;
-    }
-    return paginate(this.pagination, field, query);
+  protected paginateBy(tiebreaker: string, aggregator: string) {
+    const self = this;
```

Comment:
yeah, so this guarantees the `this` binding so funny things don't happen if you pass the returned closure to `map` or something like that. Rather than passing the full `this` reference though, as you said, we could always just capture `this.pagination` and pass that along.